### ISSUE TITLE
Improve use of temporary files

### ISSUE THREAD

**Issue by gaul** [s3fs-fuse/s3fs-fuse#678]
GCC complains about one use of `tmpnam`:

```
fdcache.o: In function `FdEntity::OpenMirrorFile()':
/home/gaul/work/s3fs-fuse/src/fdcache.cpp:761: warning: the use of `tmpnam' is dangerous, better use `mkstemp'
```

Coverity complains about four uses of `tmpfile`, recommending `mkstemp` instead:

https://scan3.coverity.com/doc/en/cov_checker_ref.html#static_checker_SECURE_TEMP

**Comment by ggtakec** [comment]
@gaul Yes, thanks.
I found another warnning lines(JSONCPP in curl.cpp, singed/unsigned in curl.cpp) in latest codes, and I will try to fix.
Please wait a moment.

**Comment by ggtakec** [comment]
Fixed by #688, thus closing this issue.
